Client/Frontend/Brave Today/Cards/AdCardView.swift

```diff
@@ -0,0 +1,107 @@
+
+class AdCardView: FeedCardBackgroundButton, FeedCardContent {
+    var actionHandler: ((Int, FeedItemAction) -> Void)?
+    var contextMenu: FeedItemMenu?
+    
+    let feedView = FeedItemView(layout: .ad).then {
+        $0.thumbnailImageView.contentMode = .scaleAspectFit
+    }
+    
+    private let adCalloutView = BraveAdCalloutView()
+    private var contextMenuDelegate: NSObject?
+    
+    required init() {
+        super.init(frame: .zero)
+        
+        addSubview(feedView)
+        feedView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        addTarget(self, action: #selector(tappedSelf), for: .touchUpInside)
+        feedView.callToActionButton.addTarget(self, action: #selector(tappedSelf), for: .touchUpInside)
+        
+        let contextMenuDelegate = FeedContextMenuDelegate(
+            performedPreviewAction: { [weak self] in
+                self?.actionHandler?(0, .opened())
+            },
+            menu: { [weak self] in
+                return self?.contextMenu?.menu?(0)
+            }
+        )
+        addInteraction(UIContextMenuInteraction(delegate: contextMenuDelegate))
+        self.contextMenuDelegate = contextMenuDelegate
+        
+        isAccessibilityElement = false
+        accessibilityElements = [feedView, feedView.callToActionButton]
+        feedView.accessibilityTraits.insert(.button)
+        shouldGroupAccessibilityChildren = true
+        
+        addSubview(adCalloutView)
+        adCalloutView.snp.makeConstraints {
+            $0.top.trailing.equalToSuperview().inset(8)
+            $0.leading.greaterThanOrEqualToSuperview().inset(8)
+            $0.bottom.lessThanOrEqualToSuperview().inset(8)
+        }
+    }
+    
+    override var accessibilityLabel: String? {
+        get { feedView.accessibilityLabel }
+        set { assertionFailure("Accessibility label is inherited from a subview: \(String(describing: newValue)) ignored") }
+    }
+    
+    @objc private func tappedSelf() {
+        actionHandler?(0, .opened())
+    }
+}
+
+private class BraveAdCalloutView: UIView {
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        backgroundColor = .white
+        layer.cornerRadius = 4
+        layer.borderColor = UIColor.braveLighterBlurple.cgColor
+        layer.borderWidth = 1
+        layer.masksToBounds = true
+        
+        let stackView = UIStackView()
+        stackView.spacing = 3
+        stackView.alignment = .center
+        stackView.layoutMargins = .init(equalInset: 5)
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.addStackViewItems(
+            .view(UIImageView(image: UIImage(imageLiteralResourceName: "bat-small")).then {
+                $0.contentMode = .scaleAspectFit
+                $0.snp.makeConstraints {
+                    $0.size.equalTo(14)
+                }
+            }),
+            .view(UILabel().then {
+                $0.text = "Ad"
```

Comment:
Good question, I'm not sure. @karenkliu @jamesmudgett do you know if "Ad" is translated in this case since its not necessarily "Brave Ads" the product. Does Desktop translate it?